sites/upsun/src/development/troubleshoot.md

```diff
@@ -118,26 +118,75 @@ If you are using SSH, see how to [troubleshoot SSH access](../development/ssh/tr
-- Certain code paths in your application are too slow and timing out.
-  - Check your code is running smoothly.
-  - Consider using the [observability solution](../increase-observability/application-metrics/_index.md) included in your project to get a better view of your application.
-- A PHP process is crashing because of a segmentation fault.
-  - See [how to deal with crashed processes](../languages/php/troubleshoot.md#troubleshoot-a-crashed-php-process).
-- A PHP process is killed by the kernel out-of-memory killer.
-  - See [how to deal with killed processes](../languages/php/troubleshoot.md#troubleshoot-a-killed-php-process).
+If you encounter a **502 Bad Gateway** or **503 Service Unavailable** error while accessing your application, it typically indicates that your application is either crashing or unavailable. Below are the typical causes and potential solutions:
+
+#### 1. Your app is listening at the wrong place
+- **Solution**: Check your app's [upstream properties](/create-apps/app-reference/single-runtime-image.md#upstream).
+- If your app is listening at a specific port, verify that it’s using the [`PORT` environment variable](/development/variables/use-variables.md#use-provided-variables) and ensure that it is listening at the correct port. 
+
+#### 2. Configuration issues in `.upsun/config.yaml`
+- **Solution**: Review your `.upsun/config.yaml` configuration.
+- Look for any errors that might prevent a process from starting or cause issues with requests being forwarded properly.
+- Check the `web.commands.start` entry or your `passthru` configuration for any misconfigurations.
+
+#### 3. Excessive traffic to your application
+- **Solution**: Check if the amount of traffic coming to your site exceeds your application's processing power.
+- Consider whether [bots or automated traffic might be overwhelming your site](https://support.platform.sh/hc/en-us/community/posts/16439634723858).
+
+#### 4. Slow code paths causing timeouts
+- **Solution**: Check your application's code to ensure it is running efficiently.
+- Use [observability tools](/increase-observability/application-metrics/_index.md) provided in your project for better insight into performance issues and potential bottlenecks.
+
+#### 5. PHP process crashing due to segmentation fault
+If your PHP process crashes due to a segmentation fault, you may encounter a message like the warning error detailed below. This indicates that either a PHP extension is causing the segmentation fault or there’s an issue in your PHP app code. 
+
+```text {location="/var/log/app.log"}
+WARNING: [pool web] child 112 exited on signal 11 (SIGSEGV) after 7.405936 seconds from start
+```
+- **Solution**: Review recent changes made to your app to identify potential causes.
+- Consider using a debugging tool such as [Xdebug](/languages/php/xdebug.md) to help with quicker troubleshooting and identify the root cause of the crash.
```

Comment:
Is this something we've identified commonly in our customers? If yes, then that's definitely the place for this to be.

My PHP experience is limited, but I wouldn't consider this a common case.